### PR TITLE
feat(auth): JWT token

### PR DESCRIPTION
**Pull request by martwozniak**
* JWT
* Server Side Cookie generation
* Now middleware require user to pass `token` and `workspace`

### CODE REVIEW

File: pages/api/public/token.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+    const token = await getToken({ req: req, secret: secret });
+  if (token) {
+    // Signed in
+    console.log("JSON Web Token", JSON.stringify(token, null, 2));
+    res.status(200).json(JSON.stringify(token, null, 2))
+  } else {
+    // Not Signed in
+    res.status(401);
```
Comment (grski) [inline pages/api/public/token.ts:15]: check my comments regarding hardcoding statuses in another PR

File: pages/api/public/auth/logout.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import cookie from "cookie";
+
+export default (req: NextApiRequest, res: NextApiResponse) => {
+  res.setHeader(
+    "Set-Cookie",
+    cookie.serialize("token", "", {
+      httpOnly: true,
+      secure: process.env.NODE_ENV !== "development",
+      expires: new Date(0),
```
Comment (grski) [inline pages/api/public/auth/logout.ts:10]: interesting 🤔 mby some expiration date will do good?

Comment (martwozniak) [reply pages/api/public/auth/logout.ts:10]: this is expiration date - set to ` new Date(0)` so default value in the past Thu Jan 01 1970 01:00:00 GMT+0100 (czas środkowoeuropejski standardowy) {}
image

**Comment by martwozniak** [comment]
Updated

File: pages/api/public/token.ts (hunk 1 above)
Comment (grski) [reply pages/api/public/token.ts:15]: still relevant

Comment (grski) [reply pages/api/public/token.ts:15]: add a comment to handle it later
I've seen the other code you did so it just needs to be changed later

**Review by grski — APPROVED** [review]
lgtm

**Pull request merged**